src/ra_server_proc.erl

```diff
@@ -1872,11 +1874,20 @@ handle_process_down(Pid, Info, RaftState,
                                     monitors = Monitors}),
     {keep_state, State, Actions}.
 
-record_leader_change(Leader, #state{conf = #conf{cluster_name = ClusterName},
-                                    server_state = ServerState}) ->
-    Members = do_state_query(members, ServerState),
-    ok = ra_leaderboard:record(ClusterName, Leader, Members),
-    ok.
+record_cluster_change(
+  #state{conf = #conf{cluster_name = ClusterName},
+         server_state = ServerStateA},
+  #state{server_state = ServerStateB}) ->
+    LeaderA = do_state_query(leader, ServerStateA),
```

Comment:
can we do something like `do_state_query(leader A) =/= do_state_query(leader, B) orelse do_state_query(members, A)....` to potentially avoid an equality check on the less likely change?